File: backend/payment-service/src/main/java/com/ecommerce/payment/service/PaymentService.java
package com.ecommerce.payment.service;

import com.ecommerce.payment.dto.request.CreatePaymentRequest;
import com.ecommerce.payment.dto.response.PayPalOrderResponse;
import com.ecommerce.payment.dto.response.PaymentResponse;

import java.util.List;

public interface PaymentService {

    PayPalOrderResponse createPayment(String email, CreatePaymentRequest request);

    PaymentResponse capturePayment(String paypalOrderId, String payerId);

    PaymentResponse cancelPayment(String paypalOrderId);

    PaymentResponse getPaymentByOrderId(Long orderId);

    List<PaymentResponse> getMyPayments(String email);

    List<PaymentResponse> getAllPayments();
}

File: frontend/react-app/src/services/paymentService.js
import axiosInstance from '../utils/axiosConfig'

const paymentService = {

  createPayment: async (orderId) => {
    const response = await axiosInstance.post('/api/v1/payments/create', {
      orderId,
    })
    return response.data
  },

  getPaymentByOrderId: async (orderId) => {
    const response = await axiosInstance.get(`/api/v1/payments/order/${orderId}`)
    return response.data
  },

  getMyPayments: async () => {
    const response = await axiosInstance.get('/api/v1/payments/my-payments')
    return response.data
  },
}

export default paymentService